finagle-netty4/src/main/scala/com/twitter/finagle/netty4/channel/ServerBridge.scala

```diff
@@ -15,6 +17,17 @@ private[netty4] class ServerBridge[In, Out](
 
   def initChannel(ch: Channel): Unit = {
     val transport: Transport[In, Out] = transportFac(ch)
-    serveTransport(transport)
```

Comment:
Oh interesting. It looks like [we change the `isRegistered` flag on a channel before firing the `channelRegistered` event](https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/AbstractChannel.java#L502-L509). You're totally right. We should probably change server bridge to be just `ChannelInboundHandler`. Is that what you had in mind?